### PR TITLE
Removed unused lockfile, Added SLES 11 support

### LINKED ISSUE #6798: RPM-Installation not complete
I'm on a `SUSE Linux Enterprise Server 11 (x86_64)` and evaluating ElasticSearch to replace an existing solution.

I downloaded [the rpm-release](https://download.elasticsearch.org/elasticsearch/elasticsearch/elasticsearch-1.2.1.noarch.rpm) and tried to install it using `rpm -i elasticsearch-1.2.1.noarch.rpm` but neither the group shown in [preinstall](https://github.com/elasticsearch/elasticsearch/blob/master/src/rpm/scripts/preinstall) is added nor the [init-script](https://github.com/elasticsearch/elasticsearch/blob/master/src/rpm/init.d/elasticsearch) is moved/copied to `/etc/init.d`.

Am I missing something or is the installation routine messed up for SLES?

### PR DESCRIPTION
**Pull request by jjfalling**
I removed the lockfile as it is not used anywhere and the service manager should take care of the locking. I also added SLES 11 support. 
Tested on RHEL 6.7 and SLES 11 SP3

This fixes #6798 and #9366

### CODE REVIEW

**Comment by electrical** [comment]
@spinscale @tlrx ping

**Comment by spinscale** [comment]
I only have a SP1 version to tests this against, and that one fails because of the RPM signature issue... no chance of automating this, unless there is a working SP3 vagrant VM

**Comment by electrical** [comment]
@spinscale SP3 will fail as well with the signed RPM.
We will need a second rpm repo with the V3 signature signed repo.

**Comment by salyh** [comment]
+1 (lot of customers with SLES 11 are complaining about start scripts not working properly). The support matrix however lists SLES as a supported OS (talking about ES 2.2)

**Comment by jjfalling** [comment]
I've updated the pr to include all of the changes in master. Should be ready to test.

File: distribution/rpm/src/main/packaging/init.d/elasticsearch [hunk 1]
```diff
@@ -116,22 +115,38 @@ start() {
...
-    retval=$?
-    echo
-    [ $retval -eq 0 ] && touch $lockfile
-    return $retval
+    # if not running, start it up here.
+
+    retval=''
+    if [ $SUSE ]; then
+        # SLES uses startproc
+        startproc -u $ES_USER $exec -p $pidfile -d -Des.default.path.home=$ES_HOME -Des.default.path.logs=$LOG_DIR -Des.default.path.data=$DATA_DIR -Des.default.path.conf=$CONF_DIR &
```
Comment (jasontedor) [inline distribution/rpm/src/main/packaging/init.d/elasticsearch]: This is passing incorrect parameters to the Elasticsearch start process; we use `-E` now.

File: distribution/rpm/src/main/packaging/init.d/elasticsearch [hunk 2]
```diff
@@ -116,22 +115,38 @@ start() {
...
+        retval=$?
+        rc_status -v
+    else
+        # For non-SLES usually something like "daemon $exec"
+        daemon --user $ES_USER --pidfile $pidfile $exec -p $pidfile -d -Des.default.path.home=$ES_HOME -Des.default.path.logs=$LOG_DIR -Des.default.path.data=$DATA_DIR -Des.default.path.conf=$CONF_DIR
```
Comment (jasontedor) [inline distribution/rpm/src/main/packaging/init.d/elasticsearch]: This stomps the previous change to the init script to pass parameters using `-E`.

Comment (jjfalling) [inline distribution/rpm/src/main/packaging/init.d/elasticsearch]: Whoops, good catch

**Comment by dakrone** [comment]
@jasontedor you were the last to comment on this, do you want to review this?

**Comment by jasontedor** [comment]
> @jasontedor you were the last to comment on this, do you want to review this?

@dakrone This is not readily reviewable and can not be merged until we have Vagrant tests for SLES.

**Comment by clintongormley** [comment]
We can't support RPM installation on SLES11-SP4 (old version of RPM, see https://github.com/elastic/elasticsearch/issues/17470) so I'm going to close this.

**Comment by TomyLobo** [comment]
Great, after 5 PRs closed in favor of newer PRs, you finally decide to not take any of these simple patches to make elasticsearch work on SLES11.
It's a shitty, old OS, yes, but some people are forced to use it, have you ever thought about that before throwing away someone's work and making people's lives needlessly more difficult?

**Comment by clintongormley** [comment]
@TomyLobo We won't pretend to support something that we can't by merging PRs that may or may not work on systems that we can't test.  On top of that, the tooling we have available in master for building RPMs do not build RPMs that are installable on SLES11.  Sorry, but doing so would make our developers' lives needlessly more difficult, and would potentially do the same for our users who are frustrated when things don't work as expected.

**Comment by TomyLobo** [comment]
No one is asking you to pretend anything.
There's a difference between pretending to support something and making your best effort without giving any guarantees.

Getting SLES is a little hard, I know, but does this even run on any SUSE linux version? openSUSE Leap 42.1, for instance, the latest lts version?
According to this article, you're supposed to use rc_status, not status:
https://en.opensuse.org/openSUSE:Packaging_init_scripts#Status_Functions

http://rpmfind.net/linux/RPM/opensuse/updates/leap/42.1/oss/x86_64/aaa_base-13.2+git20140911.61c1681-13.1.x86_64.html
This package contains openSUSE Leap 42.1's version of rc.status and it looks like this function is still called rc_status, with no alias to status.

So everything here suggests that this RPM won't run on a modern SUSE either and that if you fix it there, the init.d script will work on SLES11 and SLES12 as well.

**Comment by clintongormley** [comment]
> There's a difference between pretending to support something and making your best effort without giving any guarantees.

Unfortunately, users don't see it this way in practice.  We've been bitten by this expectation before. We only support what we can test.  We can't install the RPMs on SLES 11 so we can't support it.

> Getting SLES is a little hard, I know, but does this even run on any SUSE linux version? openSUSE Leap 42.1, for instance, the latest lts version?

We test the RPMs on openSUSE 13 and on SLES 12, see https://elasticsearch-ci.elastic.co/view/All/job/elastic+elasticsearch+master+packaging-tests/

We don't currently test on Leap as nobody has asked for it thus far.  Please open an issue if you would like us to support it.

**Comment by TomyLobo** [comment]
right, i forgot they switched to systemd with opensuse 12.1/sles12, so this script would no longer even be used on those systems.
So this really is a sles11-only problem.

**Pull request closed without merging**